crates/red_knot_module_resolver/src/typeshed.rs

```diff
@@ -21,9 +37,71 @@ mod tests {
         assert!(functools_module_stub.is_file());
 
         let mut functools_module_stub_source = String::new();
-        functools_module_stub.read_to_string(&mut functools_module_stub_source)?;
+        functools_module_stub
+            .read_to_string(&mut functools_module_stub_source)
+            .unwrap();
 
         assert!(functools_module_stub_source.contains("def update_wrapper("));
-        Ok(())
+    }
+
+    #[test]
+    fn typeshed_vfs_consistent_with_vendored_stubs() {
+        let vendored_typeshed_stubs = VendoredFileSystem::new(TYPESHED_ZIP_BYTES).unwrap();
+
+        let mut empty_iterator = true;
+        for entry in walkdir::WalkDir::new(&*VENDORED_TYPESHED_DIR).min_depth(1) {
+            empty_iterator = false;
+            let entry = entry.unwrap();
+            let absolute_path = entry.path();
+            let file_type = entry.file_type();
+
+            let relative_path = absolute_path.strip_prefix(&*VENDORED_TYPESHED_DIR).unwrap();
+            let vendored_path = VendoredPath::new(relative_path.to_str().unwrap());
+
+            assert!(vendored_typeshed_stubs.exists(vendored_path));
+
+            let vendored_path_kind = vendored_typeshed_stubs
+                .metadata(vendored_path)
+                .unwrap()
+                .kind();
+            assert_eq!(vendored_path_kind.is_directory(), file_type.is_dir());
+        }
+        assert!(!empty_iterator);
+    }
+
+    #[test]
+    fn typeshed_versions_consistent_with_vendored_stubs() {
+        const VERSIONS_DATA: &str = include_str!("../vendor/typeshed/stdlib/VERSIONS");
+        let vendored_typeshed_versions = TypeshedVersions::from_str(VERSIONS_DATA).unwrap();
+
+        let mut empty_iterator = true;
+        for entry in std::fs::read_dir(VENDORED_TYPESHED_DIR.join(TYPESHED_STDLIB_DIR)).unwrap() {
+            empty_iterator = false;
+            let entry = entry.unwrap();
+            let absolute_path = entry.path();
+            let relative_path = absolute_path
+                .strip_prefix(&*VENDORED_TYPESHED_DIR)
+                .and_then(|path| path.strip_prefix(TYPESHED_STDLIB_DIR))
+                .unwrap();
+            if relative_path == Path::new("VERSIONS") {
```

Comment:
Nit: I think `Path` implements `PartialEq<str>`
```suggestion
            if relative_path == "VERSIONS" {
```

Reply:
Compiler says no:

```
error[E0277]: can't compare `std::path::Path` with `str`
  --> crates/red_knot_module_resolver/src/typeshed.rs:86:30
   |
86 |             if relative_path == "VERSIONS" {
   |                              ^^ no implementation for `std::path::Path == str`
   |
   = help: the trait `std::cmp::PartialEq<str>` is not implemented for `std::path::Path`, which is required by `&std::path::Path: std::cmp::PartialEq<&str>`
   = help: the following other types implement trait `std::cmp::PartialEq<Rhs>`:
             <&'a std::path::Path as std::cmp::PartialEq<camino::Utf8Path>>
             <&'a std::path::Path as std::cmp::PartialEq<camino::Utf8PathBuf>>
             <&'a std::path::Path as std::cmp::PartialEq<std::borrow::Cow<'b, std::ffi::OsStr>>>
             <&'a std::path::Path as std::cmp::PartialEq<std::ffi::OsStr>>
             <&'a std::path::Path as std::cmp::PartialEq<std::ffi::OsString>>
             <&'a std::path::Path as std::cmp::PartialEq<std::path::PathBuf>>
             <&'b std::path::Path as std::cmp::PartialEq<std::borrow::Cow<'a, std::path::Path>>>
             <std::path::Path as std::cmp::PartialEq<&'a camino::Utf8Path>>
           and 9 others
   = note: required for `&std::path::Path` to implement `std::cmp::PartialEq<&str>`
```